libdnf/module/ModulePackage.cpp

```diff
@@ -0,0 +1,358 @@
+}
+
+/**
+ * @brief Return a module profile by name.
+ *
+ * @return std::shared_ptr<ModuleProfile>
+ */
+std::shared_ptr<Profile> ModulePackage::getProfile(const std::string &name) const
+{
+    return metadata->getProfile(name);
+}
+
+/**
+ * @brief Return list of ModuleProfiles.
+ *
+ * @return std::vector<std::shared_ptr<ModuleProfile>>
+ */
+std::vector<std::shared_ptr<ModuleProfile>> ModulePackage::getProfiles() const
+{
+    return metadata->getProfiles();
+}
+
+/**
+ * @brief Return list of ModuleDependencies.
+ *
+ * @return std::vector<std::shared_ptr<ModuleDependencies>>
+ */
+std::vector<std::shared_ptr<ModuleDependencies>> ModulePackage::getModuleDependencies() const
+{
+    return metadata->getDependencies();
+}
+
+/**
+ * @brief Is a ModulePackage part of an enabled stream?
+ *
+ * @return bool
+ */
+bool ModulePackage::isEnabled()
+{
+    return state == ModuleState::ENABLED;
+}
+
+/**
+ * @brief Mark ModulePackage as part of an enabled stream.
+ */
+void ModulePackage::enable()
+{
+    state = ModuleState::ENABLED;
+}
+
+/**
+ * @brief Add conflict with a module stream represented as a ModulePackage.
+ */
+void ModulePackage::addStreamConflict(const std::shared_ptr<ModulePackage> &package)
+{
+    std::ostringstream ss;
+    Solvable *solvable = pool_id2solvable(pool, id);
+
+    ss << "module(" + package->getNameStream() + ")";
+    auto depId = pool_str2id(pool, ss.str().c_str(), 1);
+    solvable_add_deparray(solvable, SOLVABLE_CONFLICTS, depId, -1);
+}
+
+static std::string getPlatformStream(const std::string &osReleasePath)
+{
+    auto file = libdnf::File::newFile(osReleasePath);
+    file->open("r");
+    std::string line;
+    while (file->readLine(line)) {
+        if (line.find("PLATFORM_ID") != std::string::npos) {
+            auto equalCharPosition = line.find('=');
+            if (equalCharPosition == std::string::npos)
+                throw std::runtime_error("Invalid format (missing '=') of PLATFORM_ID in " + osReleasePath);
+
+            auto colonCharPosition = line.find(':', equalCharPosition + 1);
+            if (colonCharPosition == std::string::npos) {
+                throw std::runtime_error("Invalid format (missing ':' in value) of PLATFORM_ID in " + osReleasePath);
+            }
+            auto endPosition = line.find('"', colonCharPosition + 1);
+            if (endPosition == std::string::npos) {
+                throw std::runtime_error("Invalid format (missing '\"' in value) of PLATFORM_ID in " + osReleasePath);
+            }
+            return line.substr(colonCharPosition + 1, endPosition - colonCharPosition -1);
+        }
+    }
+
+    throw std::runtime_error("Missing PLATFORM_ID in " + osReleasePath);
+}
+
+Id createPlatformSolvable(Pool *pool, const std::string &osReleasePath)
+{
+    Repo *repo = repo_create(pool, HY_SYSTEM_REPO_NAME);
+    Id id = repo_add_solvable(repo);
+    Solvable *solvable = pool_id2solvable(pool, id);
+
+    std::string name = "platform";
```

Comment:
platform name shouldn't be hard-coded, use whatever is specified in /etc/os-release